a3fe/run/leg.py

```diff
@@ -253,7 +253,7 @@ def setup(
             system = self.run_ensemble_equilibration(sysprep_config=cfg)
 
         # Write input files
-        self.write_input_files(system, config=cfg)
+        self.setup_stages(system, config=cfg)
```

Comment:
Maybe call config systemprep_config or something just to make it more obvious that it's not an engine_config.